pandas/core/algorithms.py

```diff
@@ -1573,9 +1580,10 @@ def searchsorted(arr, value, side="left", sorter=None) -> np.ndarray:
             dtype = value_arr.dtype
 
         if is_scalar(value):
-            value = dtype.type(value)
+            # We know that value is int
```

Comment:
i think in this case we are specifically trying to get a same-sized np.integer object (per comment L1569-L1572)

Reply:
instead of casting, could just do an isinstance check instead of is_scalar

Reply:
Have fixed with some casts in next commit

Reply:
i'm not necessarily advocating a code change, but the casts would be unnecessary if we check for array-like instead of scalar?

How is value guaranteed an `int`? `value` is typed as `NumpyValueArrayLike` and that is defined as `Union[PythonScalar, Sequence[PythonScalar]]`. It can be any Scalar type to be inserted into the array?